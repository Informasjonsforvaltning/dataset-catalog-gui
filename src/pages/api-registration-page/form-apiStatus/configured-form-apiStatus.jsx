import _ from 'lodash';
import { reduxForm } from 'redux-form';
import { FormApiStatus } from './form-apiStatus.component';
import { asyncValidate } from '../async-patch/async-patch';
import { validate } from './form-apiStatus.validations';

const config = {
  form: 'apiStatus',
  validate,
  shouldAsyncValidate: _.stubTrue, // override default, save even if sync validation fails
  asyncValidate
};

export const ConfiguredFormApiStatus = reduxForm(config)(FormApiStatus);
