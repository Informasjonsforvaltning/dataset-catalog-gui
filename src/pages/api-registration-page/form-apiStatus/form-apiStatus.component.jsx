import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import { Field } from 'redux-form';
import localization from '../../../services/localization';
import { Helptext } from '../../../components/helptext/helptext.component';
import SelectField from '../../../components/fields/field-select/field-select.component';
import DatepickerField from '../../../components/fields/field-datepicker/field-datepicker.component';
import TextAreaField from '../../../components/fields/field-textarea/field-textarea.component';
import InputField from '../../../components/fields/field-input/field-input.component';

const apiStatusSortOrder = {
  EXPERIMENTAL: 0,
  STABLE: 1,
  DEPRECATED: 2,
  REMOVED: 3
};

export const FormApiStatus = ({ apiStatusItems, apiStatusCodeFromForm }) => (
  <form>
    <div className="form-group">
      <Helptext title={localization.schema.apiStatus.helptext.apiStatus} />
      <Field
        name="statusCode"
        component={SelectField}
        items={_.sortBy(apiStatusItems, item => apiStatusSortOrder[item.code])}
        valueKey="code"
        saveObject={false}
      />

      {(apiStatusCodeFromForm === 'DEPRECATED' ||
        apiStatusCodeFromForm === 'REMOVED') && (
        <div className="mt-4">
          <div className="form-group">
            <Helptext
              title={
                localization.schema.apiStatus.helptext
                  .deprecationInfoExpirationDate
              }
            />
            <Field
              name="deprecationInfoExpirationDate"
              type="text"
              component={DatepickerField}
              label={
                localization.schema.apiStatus.helptext
                  .deprecationInfoExpirationDate
              }
            />
          </div>
          <div className="form-group">
            <Helptext
              title={
                localization.schema.apiStatus.helptext.deprecationInfoMessage
              }
            />
            <Field
              name="deprecationInfoMessage"
              component={TextAreaField}
              label={
                localization.schema.apiStatus.helptext.deprecationInfoMessage
              }
            />
          </div>
          <div className="form-group">
            <Helptext
              title={
                localization.schema.apiStatus.helptext
                  .deprecationInfoReplacedWithUrl
              }
            />
            <Field
              name="deprecationInfoReplacedWithUrl"
              component={InputField}
              label={
                localization.schema.apiStatus.helptext
                  .deprecationInfoReplacedWithUrl
              }
            />
          </div>
        </div>
      )}
    </div>
  </form>
);

FormApiStatus.defaultProps = {
  apiStatusItems: null,
  apiStatusCodeFromForm: null
};

FormApiStatus.propTypes = {
  apiStatusItems: PropTypes.array,
  apiStatusCodeFromForm: PropTypes.string
};
