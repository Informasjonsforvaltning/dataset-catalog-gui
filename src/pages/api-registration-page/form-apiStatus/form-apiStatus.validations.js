import { validateURL } from '../../../validation/validation';

export const validate = ({ deprecationInfoReplacedWithUrl }) => {
  let errors = {};

  errors = validateURL(
    'deprecationInfoReplacedWithUrl',
    deprecationInfoReplacedWithUrl,
    errors,
    true
  );

  return errors;
};
