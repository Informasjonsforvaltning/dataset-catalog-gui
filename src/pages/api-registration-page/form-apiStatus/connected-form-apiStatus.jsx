import { formValueSelector, getFormSyncErrors } from 'redux-form';
import { connect } from 'react-redux';
import _ from 'lodash';
import { ConfiguredFormApiStatus } from './configured-form-apiStatus';

const selector = formValueSelector('apiStatus');

const mapStateToProps = (state, ownProps) => {
  const { apiItem } = ownProps;
  const apiStatusCodeFromForm = selector(state, 'statusCode');
  return {
    syncErrors: getFormSyncErrors('apiStatus')(state.form),
    initialValues: {
      statusCode: _.get(apiItem, 'statusCode'),
      deprecationInfoExpirationDate: _.get(
        apiItem,
        'deprecationInfoExpirationDate'
      ),
      deprecationInfoMessage: _.get(apiItem, 'deprecationInfoMessage'),
      deprecationInfoReplacedWithUrl: _.get(
        apiItem,
        'deprecationInfoReplacedWithUrl'
      )
    },
    apiStatusCodeFromForm
  };
};

export const ConnectedFormApiStatus = connect(mapStateToProps)(
  ConfiguredFormApiStatus
);
