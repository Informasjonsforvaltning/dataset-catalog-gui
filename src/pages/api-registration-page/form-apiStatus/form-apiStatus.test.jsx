import React from 'react';
import { shallow } from 'enzyme';
import { FormApiStatus } from './form-apiStatus.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {};
  wrapper = shallow(<FormApiStatus {...defaultProps} />);
});

test('should render FormApiStatus, { renderLandingpage  correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
