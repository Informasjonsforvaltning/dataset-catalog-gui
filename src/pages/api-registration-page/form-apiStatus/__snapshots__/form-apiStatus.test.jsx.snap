// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormApiStatus, { renderLandingpage  correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      title="Status pÃ¥ API-et"
    />
    <Field
      component={[Function]}
      items={Array []}
      name="statusCode"
      saveObject={false}
      valueKey="code"
    />
  </div>
</form>
`;
